### DISCUSSION TITLE
Public API for extracting lineage info

### DISCUSSION THREAD

**Discussion by antoniivanov** [reata/sqllineage#243] [Q&A]
Hi, 

I think the project looks great and it's very promising as a generic way to be able to gather lineage for SQL users. 

But is it only Command-Line? Are there public classes that can be used as a library in another project?

**Comment by antoniivanov** [comment]
Is LineageRunner considered a public API - https://sqllineage.readthedocs.io/en/latest/basic_concepts/lineage_runner.html 

If so then I guess my next question is if I import it then this (https://github.com/reata/sqllineage/blob/master/sqllineage/__init__.py#L59) overrides my logging configuration? Should we fix it? I can open an issue or even a PR for that.

Thanks.

**Reply by reata** [reply]
Thanks for reaching out.

Moving towards a Python library is definitely the way to go for sqllineage. It's one of the biggest features I'd like to bring in v1.4. But at present, I think I still need more time working on various edge case bugfix release in v1.3.x. So if you'd like to start with the logging configuration, you're more than welcome to.

As for the LineageRunner, I'm not very sure yet. This is definitely the most stable class among others, of course. But previously I was prioritizing adding some functional interface first (#226) . If we're doing it, should `sqllineage.run` just return `LineageRunner`? Let me know your thoughts.

**Reply by antoniivanov** [reply]
Got it. Thanks for the answer. 

I opened https://github.com/reata/sqllineage/pull/245 I hope it's ok.

**Reply by reata** [reply]
I just merged this PR, thanks for your contribution.

**Reply by antoniivanov** [reply]
Thanks . If you are curious, I am using sqllineage [here](https://github.com/vmware/versatile-data-kit/pull/783/files#diff-2e91a5548e7478c0aad99122a2de6998e976f6403b34f5a3bf49c9727cca662c)
